lawrencium/src/main/bash/run_job.sh

```diff
@@ -0,0 +1,144 @@
+#!/bin/bash
+
+# This is the script to run BEAM simulation on Lawrencium cluster.
+# By-default it expects 2 input arguments: <run name> <expected execution time>,
+# though both might be manually filled in the script body.
+
+CODE_PHRASE="Execute the body of the job."
+
+# Doing shell magic - the script will send itself as a job to the cluster,
+# and first argument will tell what to do - start a job or execute the job body.
+# This way we could avoid using two shell scripts - one to start a job and another as job body.
+# if the first argument is not what we are looking for - then this shell script is used to start a job
+if [[ "$1" != "$CODE_PHRASE" ]]; then
+  echo "Starting the job .."
+
+  # what code, data and config to use for simulation
+  export BEAM_BRANCH_NAME="develop"
+  export BEAM_COMMIT_SHA
+  export BEAM_DATA_BRANCH_NAME="develop"
+  export BEAM_DATA_COMMIT_SHA
+  export BEAM_CONFIG="test/input/beamville/beam.conf"
+  export PROFILER # either empty, 'cpu' or 'cpumem'
+
+  PARTITION="es1"
+  QOS="es_normal"
+  MEMORY_LIMIT="480"  ## in G
+
+  # if uploading to s3 required - both AWS key parts should be set
+  export S3_REGION="us-east-2"
+  export S3_PUBLISH="false"
+  export AWS_SECRET_ACCESS_KEY
+  export AWS_ACCESS_KEY_ID
+
+  # for sending notifications to slack SLACK_HOOK required
+  # for sending updates to the spreadsheet SIMULATIONS_SPREADSHEET_UPDATE_URL required
+  export SEND_NOTIFICATION="false"
+  export SLACK_HOOK_WITH_TOKEN
+  export SIMULATIONS_SPREADSHEET_UPDATE_URL
+
+  ACCOUNT="pc_beamcore"
+
+  # INPUT Argument #1 - run name
+  RUN_NAME="$1"
+  # INPUT Argument #2 - expected simulation duration
+  EXPECTED_TIME="$2" # D-HH:MM:SS, i.e. for 1 day, 2 hours and 30 minutes => 1-02:30:00
+
+  # required for doing speed comparison (BEAM simulation vs Google observations)
+  export GOOGLE_API_KEY
+
+  if [[ -z "$MEMORY_LIMIT" ]]; then
+    echo "Error: MEMORY_LIMIT is not set."
+    exit 1
+  else
+    # using the current memory limit as MAX RAM for BEAM simulation
+    export MAX_RAM="$MEMORY_LIMIT"
+  fi
+
+  if [[ -z "$RUN_NAME" ]]; then
+    echo "Error: RUN_NAME is not set."
+    exit 1
+  else
+    # adding current user name as part of simulation title
+    export TITLED="$USER/$RUN_NAME"
```

Comment:
$USER is expected to be set in lawrencium?

Reply:
Yes, it is set by cluster login node